R/derive_date_vars.R

```diff
@@ -1534,22 +1558,58 @@ derive_vars_dtm <- function(dataset,
     values = c("auto", "both", "date", "time", "none"),
     case_sensitive = FALSE
   )
+  if ((highest_imputation == "Y" & is.null(min_dates) & is.null(max_dates)) |
+    (highest_imputation == "Y" & length(min_dates) == 0 & length(max_dates) == 0)) {
+    stop("If `highest_impuation` = \"Y\" is specified, `min_dates` or `max_dates` should be specified respectively.") # nolint
+  }
 
   dtm <- paste0(new_vars_prefix, "DTM")
 
   # Issue a warning if --DTM already exists
   warn_if_vars_exist(dataset, dtm)
   mask <- rlang::as_data_mask(dataset)
-  dataset[[dtm]] <- convert_dtc_to_dtm(
-    dtc = eval_tidy(dtc, mask),
-    highest_imputation = highest_imputation,
-    date_imputation = date_imputation,
-    time_imputation = time_imputation,
-    min_dates = lapply(min_dates, eval_tidy, data = mask),
-    max_dates = lapply(max_dates, eval_tidy, data = mask),
-    preserve = preserve
-  )
 
+  if (highest_imputation == "Y" &
+    (is.null(min_dates) | length(min_dates) == 0) &
+    !is.null(max_dates)) {
+    if (date_imputation != "last") {
+      stop("If `highest_impuation` = \"Y\" and `max_dates` is specified, input `date_imputation` = \"last\"") # nolint
+    }
+    dataset[[dtm]] <- convert_dtc_to_dtm(
+      dtc = eval_tidy(dtc, mask),
+      highest_imputation = highest_imputation,
+      date_imputation = date_imputation,
+      time_imputation = time_imputation,
+      min_dates = NULL,
+      max_dates = lapply(max_dates, eval_tidy, data = mask),
+      preserve = preserve
+    )
+  } else if (highest_imputation == "Y" &
+    !is.null(min_dates) &
+    (is.null(max_dates) | length(max_dates))) {
+    if (date_imputation != "first") {
+      stop("If `highest_impuation` = \"Y\" and `min_dates` is specified, input `date_imputation` = \"first\"") # nolint
+    }
+    dataset[[dtm]] <- convert_dtc_to_dtm(
+      dtc = eval_tidy(dtc, mask),
+      highest_imputation = highest_imputation,
+      date_imputation = date_imputation,
+      time_imputation = time_imputation,
+      min_dates = lapply(min_dates, eval_tidy, data = mask),
+      max_dates = NULL,
+      preserve = preserve
+    )
+  } else {
+    dataset[[dtm]] <- convert_dtc_to_dtm(
+      dtc = eval_tidy(dtc, mask),
+      highest_imputation = highest_imputation,
+      date_imputation = date_imputation,
+      time_imputation = time_imputation,
+      min_dates = lapply(min_dates, eval_tidy, data = mask),
```

Comment:
@bundfussr think the additions above, fixed what was in your previous comment but I can't imagine this is the cleanest solution, wondering if I can run some sort of 
`if_else(is.null(min_dates), NULL, lapply(...))` statement here